test_ensure_mkdir_p: OSError: [Errno 36] File name too long

https://travis-ci.org/jayvdb/vistir/jobs/504095856 Python 3.6

```console
_____________________________ test_ensure_mkdir_p ______________________________
    @given(legal_path_chars(), legal_path_chars())
>   @example(base_dir=u'0', subdir=u'\x80')
    @settings(suppress_health_check=(HealthCheck.filter_too_much,))
    def test_ensure_mkdir_p(base_dir, subdir):
tests/test_path.py:49: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_path.py:62: in test_ensure_mkdir_p
    target = join_with_dir(base_dir, subdir)
src/vistir/path.py:255: in decorated
    mkdir_p(path, mode=mode)
src/vistir/path.py:244: in mkdir_p
    os.makedirs(os.path.join(head, tail), mode)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
name = b'/tmp/tmp_nab54vs/0/\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\...80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80'
mode = 511, exist_ok = False
    def makedirs(name, mode=0o777, exist_ok=False):
        """makedirs(name [, mode=0o777][, exist_ok=False])
    
        Super-mkdir; create a leaf directory and all intermediate ones.  Works like
        mkdir, except that any intermediate path segment (not just the rightmost)
        will be created if it does not exist. If the target directory already
        exists, raise an OSError if exist_ok is False. Otherwise no exception is
        raised.  This is recursive.
    
        """
        head, tail = path.split(name)
        if not tail:
            head, tail = path.split(head)
        if head and tail and not path.exists(head):
            try:
                makedirs(head, mode, exist_ok)
            except FileExistsError:
                # Defeats race condition when another thread created the path
                pass
            cdir = curdir
            if isinstance(tail, bytes):
                cdir = bytes(curdir, 'ASCII')
            if tail == cdir:           # xxx/newdir/. exists if xxx/newdir exists
                return
        try:
>           mkdir(name, mode)
E           OSError: [Errno 36] File name too long: b'/tmp/tmp_nab54vs/0/\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80\xf0\x90\x80\x80'
../../../virtualenv/python3.6.3/lib/python3.6/os.py:220: OSError
```

--- a/news/65.bugfix.rst
+++ b/news/65.bugfix.rst
@@ -1,0 +1,1 @@
+Fixed a bug which caused test failures due to generated paths on *nix based operating systems which were too long.

--- a/tests/test_path.py
+++ b/tests/test_path.py
@@ -38,6 +38,7 @@ def test_mkdir_p(base_dir, subdir):
     assume(not any((dir_name in ["", ".", "./", ".."] for dir_name in [base_dir, subdir])))
     assume(not (os.path.relpath(subdir, start=base_dir) == "."))
     assume(os.path.abspath(base_dir) != os.path.abspath(os.path.join(base_dir, subdir)))
+    assume(len(base_dir) < 255 and len(subdir) < 255)
     with vistir.compat.TemporaryDirectory() as temp_dir:
         target = os.path.join(temp_dir.name, base_dir, subdir)
         assume(vistir.path.abspathu(target) != vistir.path.abspathu(os.path.join(temp_dir.name, base_dir)))

